src/passes/CodePushing.cpp

```diff
@@ -323,6 +323,26 @@ class Pusher {
       assert(i > 0);
       i--;
       auto* pushable = isPushable(list[i]);
+      if (pushable && pushable->type == Type::unreachable) {
+        // Don't try to push something unreachable. If we did, then we'd need to
+        // refinalize the block we are moving it from:
```

Comment:
Is having to refinalize a problem? Are there any other cases in this pass that we need to refinalize?

Reply:
It's possible to refinalize. In this pass though we don't even include the header for it - nothing else required it - so it would definitely be more work to add refinalizing logic than not (specially if we try to refinalize only when necessary, to be more efficient). And in general leaving work for DCE is reasonable, I think.